### PR TITLE
Windows: Increase wait timer resolution

### PR DESCRIPTION
**Pull request by msiglreith**
Addresses #2005 

Temporarily increase the timer resolution for the wait operation. By default it's around 16ms which is too long for most realtime applications and leads to unpredictable results. A future enhancement in this direction could be allow the user to control it. Possible use case would be UI application with some animations running with minimal energy impact.

Reported by https://twitter.com/olson_dan/status/1428840173354160128?s=20
With the given change for the wgpu `water` example the framerate is much closer to the target of 16.6ms on my system.

Before
```
Avg frame time 28.707153ms
Avg frame time 28.356724ms
Avg frame time 28.303003ms
Avg frame time 28.835732ms
Avg frame time 29.122643ms
Avg frame time 27.729534ms
```

After
```
Avg frame time 16.845396ms
Avg frame time 16.859407ms
Avg frame time 16.840519ms
Avg frame time 16.862593ms
Avg frame time 16.856462ms
Avg frame time 16.870092ms
Avg frame time 16.868887ms
```

- [x] Tested on all platforms changed
- [x] Compilation warnings were addressed
- [x] `cargo fmt` has been run on this branch
- [ ] `cargo doc` builds successfully
- [x] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [ ] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [ ] Created or updated an example program if it would help users understand this functionality
- [ ] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

File: src/platform_impl/windows/event_loop.rs [hunk 1]
```diff
@@ -25,7 +25,9 @@ use winapi::{
         windowsx, winerror,
     },
     um::{
-        libloaderapi, ole2, processthreadsapi, winbase,
+        libloaderapi,
+        mmsystem::TIMECAPS,
```
Comment (filnet) [inline src/platform_impl/windows/event_loop.rs:29]: i suggest using mssystem not mmsystem::TIMECAPS.
seems to be the approach used here.

File: src/platform_impl/windows/event_loop.rs [hunk 2]
```diff
@@ -324,6 +326,17 @@ fn get_wait_thread_id() -> DWORD {
     }
 }
 
+lazy_static! {
+    static ref WAIT_PERIOD_MIN: UINT = unsafe {
+        let mut tc = TIMECAPS {
+            wPeriodMin: 0,
+            wPeriodMax: 0,
+        };
+        timeapi::timeGetDevCaps(&mut tc as *mut _, mem::size_of::<TIMECAPS>() as _);
```
Comment (filnet) [inline src/platform_impl/windows/event_loop.rs:335]: check and handle errors ?
if the call errors out, then wPeriodMin could be undefined.

File: src/platform_impl/windows/event_loop.rs [hunk 3]
```diff
@@ -366,16 +382,26 @@ fn wait_thread(parent_thread_id: DWORD, msg_window_id: HWND) {
             if let Some(wait_until) = wait_until_opt {
                 let now = Instant::now();
                 if now < wait_until {
-                    // MsgWaitForMultipleObjects tends to overshoot just a little bit. We subtract
-                    // 1 millisecond from the requested time and spinlock for the remainder to
-                    // compensate for that.
+                    // Windows' scheduler has a default accuracy of several ms. This isn't good enough for
+                    // `WaitUntil`, so we request the Windows' scheduler to use a higher accuracy if possible.
```
Comment (maroider) [inline src/platform_impl/windows/event_loop.rs:386]:
```suggestion
                    // `WaitUntil`, so we request the Windows scheduler to use a higher accuracy if possible.
```
nit: This sentence feels weird. Removing the apostrophe makes it feel less so.

File: CHANGELOG.md [hunk 4]
```diff
@@ -15,6 +15,8 @@
 - On Wayland, implement `request_user_attention` with `xdg_activation_v1`.
 - On X11, emit missing `WindowEvent::ScaleFactorChanged` when the only monitor gets reconnected.
 
+- On Windows, increase wait timer resolution for more accurate timing when using `WaitUntil`.
+
```
Comment (maroider) [inline CHANGELOG.md:19]: Idk why, but there's an extra newline here.

**Review by maroider — CHANGES_REQUESTED** [review]
I did a once-over and marked everything that looked to be resolved as "resolved".
I only have a couple of nitpicks.

**Comment by msiglreith** [comment]
Thanks for reviewing again, addressed the remaining parts

**Review by maroider — APPROVED** [review]

**Pull request merged**